Print extended hardware information when running in verbose mode

Calinou opened the pull request:
This is useful for troubleshooting, especially for player-submitted reports in exported projects. This complements https://github.com/godotengine/godot-proposals/issues/4732, but a dedicated dialog should still be added to the editor as not every user can easily run Godot from a terminal (e.g. Android/HTML5).

This is the generated output on my system when running with `--verbose`:

```
System information:

Operating system: Fedora Linux 42 (KDE Plasma Desktop Edition)
Device model: GenericDevice
CPU model: 13th Gen Intel(R) Core(TM) i9-13900K (32 threads)
Total memory size: 62.54 GiB
GPU driver version: nvidia; 570.133.07
Audio driver: PulseAudio
Audio output mix rate: 44100 Hz
Audio output devices:
  #0: Default (selected)
  #1: alsa_output.pci-0000_01_00.1.hdmi-stereo
  #2: alsa_output.usb-Sony_Interactive_Entertainment_DualSense_Wireless_Controller-00.analog-surround-40
Audio input mix rate: 44100 Hz
Audio input devices:
  #0: Default
  #1: alsa_input.usb-C-Media_Electronics_Inc._USB_Advanced_Audio_Device-00.analog-stereo
  #2: alsa_input.usb-Sony_Interactive_Entertainment_DualSense_Wireless_Controller-00.analog-stereo
Input MIDI devices: (none)
Connected gamepads:
  #0: PS5 Controller
    { "vendor_id": 1356, "product_id": 3302, "raw_name": "Sony Interactive Entertainment DualSense Wireless Controller" }
Tablet driver: default
Displays:
  #0: (0, 0), 3840x2160, 119.88 Hz, 60 dpi, scale factor: 1.00, usable rect: (0, 0), 3840x2160

Engine information:

Engine compiled for architecture: x86_64
2D physics server: GodotPhysics2D
3D physics server: GodotPhysics3D
```

Calinou committed:
Print extended hardware information when running in verbose mode

This is useful for troubleshooting, especially for player-submitted
reports in exported projects.

bruvzg commented on main/main.cpp:
```diff
@@ -3164,6 +3164,45 @@ Error Main::setup2(bool p_show_boot_logo) {
+		}
+		print_line("Displays:");
+		for (int i = 0; i < DisplayServer::get_singleton()->get_screen_count(); i++) {
+			const Rect2i usable_rect = DisplayServer::get_singleton()->screen_get_usable_rect();
+			print_line(
+					vformat("  #%d: (%d, %d), %dx%d, %.2f Hz, %d dpi, scale factor: %.2f (maximum: %.2f), usable rect: (%d, %d), %dx%d",
+							i,
+							DisplayServer::get_singleton()->screen_get_position().x,
+							DisplayServer::get_singleton()->screen_get_position().y,
+							DisplayServer::get_singleton()->screen_get_size().x,
+							DisplayServer::get_singleton()->screen_get_size().y,
+							DisplayServer::get_singleton()->screen_get_refresh_rate(),
+							DisplayServer::get_singleton()->screen_get_dpi(),
+							DisplayServer::get_singleton()->screen_get_scale(),
+							DisplayServer::get_singleton()->screen_get_max_scale(),
```
Missing display index for all `DisplayServer::get_singleton()->screen_get_` calls (except `_max_scale`), it's printing first display data for all displays.

Calinou replied:
Good catch :slightly_smiling_face: 

I've removed the max_scale report since it's just an amalgamation of all reported screen sizes anyway (I thought it was something different initially).

MJacred commented:
@Calinou: I believe we can add info on connected [midi inputs](https://docs.godotengine.org/en/stable/classes/class_os.html#class-os-method-get-connected-midi-inputs) and [joypads](https://docs.godotengine.org/en/stable/classes/class_input.html#class-input-method-get-connected-joypads) (names; note if no button mapping).  
It should be noted that the way the Joypad name is retrieved could change, if it's ok for me to [change the output for xinput device names](https://github.com/godotengine/godot/pull/98861#issuecomment-2470513052) (which would make the (end user's) code logic consistent across platforms). And the [graphics card info](https://docs.godotengine.org/en/stable/classes/class_renderingserver.html#class-renderingserver-method-get-video-adapter-name).

Calinou commented:
@MJacred I've added MIDI inputs and GPU driver version to this PR. (The GPU model name is already stated in the header printed on every startup even in non-verbose mode, so I didn't repeat it.)

There's the question of whether we should move the GPU driver version to the non-verbose header, but it impacts startup times when queried the first time (especially on Linux), so I'd prefer not until we find a way to address this slowdown.

MJacred commented:
> There's the question of whether we should move the GPU driver version to the non-verbose header, but it impacts startup times when queried the first time (especially on Linux), so I'd prefer not until we find a way to address this slowdown.

Sounds reasonable to leave it out (unless the slowdown can be addressed).  
Personally, I don't think we need to worry about this. Eventually, we'll get do that editor menu, where all the important software and hardware info is listed.

Calinou commented:
This PR crashed when using the project manager on startup due to MIDI inputs on Linux (https://github.com/godotengine/godot/issues/52821):

```
#0  0x0000000000000000 in ?? ()
#1  0x00000000060e3c9d in MIDIDriverALSAMidi::open (this=0x7fffffffc718) at drivers/alsamidi/midi_driver_alsamidi.cpp:78
#2  0x000000000a889e57 in OS::open_midi_inputs (this=0x7fffffffc4c0) at ./core/os/os.cpp:647
#3  0x000000000341c3ce in Main::setup2 (p_show_boot_logo=true) at main/main.cpp:3435
#4  0x000000000341752a in Main::setup (execpath=0x7fffffffd0f9 "/home/hugo/Documents/Git/godotengine/godot/bin/godot.linuxbsd.editor.x86_64", 
    argc=1, argv=0x7fffffffcb30, p_second_phase=true) at main/main.cpp:2789
#5  0x0000000003316965 in main (argc=2, argv=0x7fffffffcb28) at platform/linuxbsd/godot_linuxbsd.cpp:74
```

This is because the project manager uses the Dummy audio driver, so I've added a workaround that doesn't list MIDI devices while using the Dummy audio driver.

I've also added more information that's printed on startup (see OP for an updated example).
